third_party/move/move-vm/runtime/src/data_cache.rs

```diff
@@ -219,28 +221,48 @@ impl<'r> TransactionDataCache<'r> {
                 },
             };
             // TODO(Gas): Shall we charge for this?
-            let (ty_layout, has_aggregator_lifting) =
-                loader.type_to_type_layout_with_identifier_mappings(ty, module_store)?;
-
-            let module = module_store.module_at(&ty_tag.module_id());
-            let metadata: &[Metadata] = match &module {
-                Some(module) => &module.module().metadata,
-                None => &[],
-            };
-
-            // If we need to process aggregator lifting, we pass type layout to remote.
-            // Remote, in turn ensures that all aggregator values are lifted if the resolved
-            // resource comes from storage.
-            let (data, bytes_loaded) = self.remote.get_resource_bytes_with_metadata_and_layout(
-                &addr,
-                &ty_tag,
-                metadata,
-                if has_aggregator_lifting {
-                    Some(&ty_layout)
-                } else {
-                    None
+            let (ty_layout, has_aggregator_lifting) = loader
+                .type_to_type_layout_with_identifier_mappings(ty, module_store, module_storage)?;
+
+            let (data, bytes_loaded) = match loader {
+                Loader::V1(_) => {
+                    let maybe_module = module_store.module_at(&ty_tag.module_id());
```

Comment:
matter of style, but as a fancy fan got to mention we can directly compute the metadata by chaining .map_or_else(|| &[], |m|.module().metadata());
on one hand maybe_module isn't used elsewhere, on the other hand it makes reading a bit clearer. Similarly match might be more verbose - so up to you.